lib/clouds/libcloud_common.py

```diff
@@ -542,13 +630,17 @@ def get_ip_address(self, obj_attr_list) :
             _status = 100
             node = self.get_instances(obj_attr_list)
 
-            if len(node.private_ips) > 0 and obj_attr_list["run_netname"].lower() == "private" :
-                obj_attr_list["run_cloud_ip"] = node.private_ips[0]
+            if len(node.private_ips) > 0 : 
```

Comment:
As we were discussing on mattermost, we really have to stop the key/value bleeding. I'm totally fine with REMOVING all of these `use_xxxx` statements that I introduced, so long as we find a way to do this in a configuration file templates that is shared amongst all the libcloud-based adapters without requiring the user to have to define them all or even know about them all in advance to get started developing a new cloud adapter.

Reply:
I don't think that I got that... "run_netname" and "prov_netname" are common keys for all clouds, and I depend on then heavily, specially on "non-VM" clouds, such as Kubernetes, Docker and LXD. Removing these would require, in my evaluation, a significant re-design of all adapters.

Reply:
OK, so there was some detection logic in get_ip_address that was removed for making sure that `run_cloud_ip` points to the VPN IP address instead of the other IP address. What happened to it?